### PR TITLE
doc: add hardware section

### PR DESCRIPTION
**Pull request by xiang90**
/cc @heyitsanthony @joshix

### CODE REVIEW

File: Documentation/docs.md [hunk 1]
```diff
@@ -26,6 +26,7 @@ Administrators who need to create reliable and scalable key-value stores for the
  - [Setting up etcd gateways][gateway]
  - [Setting up etcd gRPC proxy (pre-alpha)][grpc_proxy]
  - [Run etcd clusters inside containers][container]
+ - [Choose hardwares][hardware]
```
Comment (heyitsanthony) [inline Documentation/docs.md]: `Choosing hardware`? `Hardware recommendations`?

File: Documentation/op-guide/hardware.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
+# Choose hardwares
+
+etcd usually runs well with limited resources for development or testing purpose; it’s common to develop with etcd on a  laptop or a cheap cloud machine. However, when running etcd clusters in production, some hardware guidelines are useful for proper administration. These suggestions are not hard rules; they serve as a good starting point for a robust production deployment. As always, deployments should be tested with simulated workloads before running in production.
```
Comment (heyitsanthony) [inline Documentation/op-guide/hardware.md]: s/purpose/purposes

File: Documentation/op-guide/hardware.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+
+## Disks
+
+Fast disks are the most critical factor for etcd deployment performance and stability. 
+
+A slow disk will increase etcd request latency and potentially hurt cluster stability. Since etcd’s consensus protocol depends on persistently storing metadata to a log, a majority of etcd cluster members must write every request down to disk. Additionally, etcd will also incrementally checkpoint its state to disk so it can truncate this log. If these writes take too long, heartbeats may time out and trigger an election, undermining the stability of the cluster.
+
+etcd is very sensitive to disk write latency. Typically 50 sequential IOPS (e.g., a 7200 RPM disk) is required. For heavily load clusters, 500 sequential IOPS (e.g., a typical local SSD or a high performance virtualized block device) is recommended. Note that most cloud providers publish concurrent IOPS rather than sequential IOPS; the published concurrent IOPS can be 10x greater than the sequential IOPS. To measure actual sequential IOPS, we suggest using a disk benchmarking tool such as [diskbench][diskbench], [fio][fio].
```
Comment (heyitsanthony) [inline Documentation/op-guide/hardware.md]: s/load/loaded

Comment (heyitsanthony) [inline Documentation/op-guide/hardware.md]: s/, fio/or fio

File: Documentation/op-guide/hardware.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+etcd requires only modest disk bandwidth but more disk bandwidth buys faster recovery times when a failed member has to catch up with the cluster. Typically 10MB/s will recover 100MB data within 15 seconds. For large clusters, 100MB/s or higher is suggested for recovering 1GB data within 15 seconds.
+
+When possible, back etcd’s storage with a SSD. A SSD usually provides lower write latencies and with less variance than a spinning disk, thus improving the stability and reliability of etcd. If using spinning disk, get the fastest disks possible (15,000 RPM). Using RAID 0 is also an effective way to increase disk speed, for both spinning disks and SSD. With at least three cluster members, mirroring and/or parity variants of RAID are unnecessary; etcd consistent replication already gets high availability.
```
Comment (heyitsanthony) [inline Documentation/op-guide/hardware.md]: s/etcd consistent/etcd's consistent/

File: Documentation/op-guide/hardware.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+|----------|------|-------|--------|------|----------------|
+| AWS | m4.xlarge | 4 | 16 | 6000 | 93.75 |
+| GCE | n1-standard-4 + 150GB PD SSD | 4 | 15 | 4500 | 75 |
+
+
+### Large cluster
+
+A large cluster serves fewer than 1500 clients, serves fewer than 10,000 of requests per second, and stores no more  than 1GB of data.
+
+Example application workload: A 1000 nodes Kubernetes cluster
```
Comment (gyuho) [inline Documentation/op-guide/hardware.md]: `A 1000-node Kubernetes`?

File: Documentation/op-guide/hardware.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+| Provider | Type | vCPUs | Memory (GB) | Max concurrent IOPS | Disk bandwidth (MB/s) |
+|----------|------|-------|--------|------|----------------|
+| AWS | m4.2xlarge | 8 | 32 | 8000 | 125 |
+| GCE | n1-standard-8 + 250GB PD SSD | 8 | 30 | 7500 | 125 |
+
+
+### xLarge cluster
+
+An xLarge cluster serves more than 1500 clients, serves more than 10,000 of requests per second, and stores more than 1GB data.
```
Comment (gyuho) [inline Documentation/op-guide/hardware.md]: Just `more than 10,000 requests`?

**Comment by xiang90** [comment]
@heyitsanthony @gyuho All fixed.

**Comment by gyuho** [comment]
lgtm. defer to @heyitsanthony

**Comment by heyitsanthony** [comment]
lgtmt

**Pull request merged**